piker/brokers/kraken.py

```diff
@@ -162,6 +234,134 @@ async def _public(
         )
         return resproc(resp, log)
 
+    async def _private(
+        self,
+        method: str,
+        data: dict,
+        uri_path: str
+    ) -> Dict[str, Any]:
+        headers = {
+            'Content-Type': 
+                'application/x-www-form-urlencoded',
+            'API-Key':
+                self._api_key,
+            'API-Sign':
+                get_kraken_signature(uri_path, data, self._secret)
+        }
+        resp = await self._sesh.post(
+            path=f'/private/{method}',
+            data=data,
+            headers=headers,
+            timeout=float('inf')
+        )
+        return resproc(resp, log)
+
+    async def endpoint(
+        self,
+        method: str,
+        data: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        uri_path = f'/0/private/{method}'
+        data['nonce'] = str(int(1000*time.time()))
+        return await self._private(method, data, uri_path)
+
+    async def get_positions(
+        self,
+        data: Dict[str, Any] = {}
+    ) -> (Dict[str, Any], Dict[str, Any]):
+        data['ofs'] = 0
+        positions = {}
+        vols = {}
+        # Grab all trade history
+        # https://docs.kraken.com/rest/#operation/getTradeHistory
+        while True:
+            resp = await self.endpoint('TradesHistory', data)
+            # grab the first 50 trades
+            if data['ofs'] == 0:
+                trades = resp['result']['trades']
+            # load the next 50 trades using dict constructor
+            # for speed
+            elif data['ofs'] == 50:
+                trades = dict(trades, **resp['result']['trades'])
```

Comment:
consider this new 3.9+ merge syntax for `dict`:
https://docs.python.org/3/whatsnew/3.9.html#dictionary-merge-update-operators

Reply:
#290